frontend/webapp/containers/main/actions/choose-action-body/custom-fields/add-cluster-info.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useMemo } from 'react';
+import styled from 'styled-components';
+import { KeyValueInputsList, Text } from '@/reuseable-components';
+
+const FieldWrapper = styled.div`
+  width: 100%;
+  margin: 8px 0;
+`;
+
+const FieldTitle = styled(Text)`
+  margin-bottom: 12px;
+`;
+
+type Props = {
+  value: string;
+  setValue: (value: string) => void;
+};
+
+type Parsed = {
+  clusterAttributes: {
+    attributeName: string;
+    attributeStringValue: string;
+  }[];
+};
+
+const AddClusterInfo: React.FC<Props> = ({ value, setValue }) => {
+  const mappedValue = useMemo(
+    () =>
+      value ? (JSON.parse(value) as Parsed).clusterAttributes.map((obj) => ({ key: obj.attributeName, value: obj.attributeStringValue })) : undefined,
+    [value]
+  );
+
+  const handleChange = (
+    arr: {
+      key: string;
+      value: string;
+    }[]
+  ) => {
+    const payload: Parsed = {
```

Comment:
you can initial this array directly with map function for cleaner code:
` const payload: Parsed = {
    clusterAttributes: arr.map(({key, value}) => ({
      attributeName: key,
      attributeStringValue: value,
    })),
  };`